File: Trackfield/app/src/main/java/com/example/trackfield/dialogs/TextDialog.java
package com.example.trackfield.dialogs;

import android.app.AlertDialog;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;
import androidx.fragment.app.FragmentManager;

import com.example.trackfield.R;

public class TextDialog extends BaseDialog {

    protected DialogListener listener;

    // arguments
    private String text, hint;

    // bundle
    private final static String BUNDLE_TEXT = "text";
    private final static String BUNDLE_HINT = "hint";

    private final static String TAG_DEFAULT = "textDialog";

    ////

    public static TextDialog newInstance(String title, String message, String text, String hint, @StringRes int posBtnTxtId, String tag) {

        TextDialog instance = new TextDialog();
        Bundle bundle = putBundleBase(title, message, posBtnTxtId, tag);

        bundle.putString(BUNDLE_TEXT, text);
        bundle.putString(BUNDLE_HINT, hint);

        instance.setArguments(bundle);

        return instance;
    }

    // on

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        try {
            listener = (DialogListener) context;
        }
        catch (ClassCastException e) {
            throw new ClassCastException("Activity must implement DialogListener");
        }
    }

    // extends

    @Override
    protected void unpackBundle() {

        Bundle bundle = unpackBundleBase(TAG_DEFAULT);

        if (bundle != null) {
            text = bundle.getString(BUNDLE_TEXT, "");
            hint = bundle.getString(BUNDLE_HINT, "");
        }

    }

    @Override
    protected AlertDialog buildDialog() {

        final View dialogView = inflater.inflate(R.layout.dialog_text, null);
        final EditText et = dialogView.findViewById(R.id.editText_textField);

        et.setText(text);
        et.setHint(hint);
        if (!message.equals("")) builder.setMessage(message);

        builder.setView(dialogView).setTitle(title)
                .setPositiveButton(posBtnTxtId, (dialog, id) -> {
                    final String input = et.getText().toString();
                    listener.onTextDialogPositiveClick(input, tag);
                })
                .setNegativeButton(negBtnTxtId, (dialog, id) -> getDialog().cancel());

        return builder.show();
    }

    // interface

    public interface DialogListener {
        void onTextDialogPositiveClick(String input, String tag);
    }

}

File: Trackfield/android/src/main/java/com/felwal/android/widget/dialog/TextDialog.kt
package com.felwal.android.widget.dialog

import android.content.Context
import android.os.Bundle
import android.view.inputmethod.EditorInfo
import androidx.annotation.StringRes
import androidx.appcompat.app.AlertDialog
import com.felwal.android.R
import com.felwal.android.databinding.DialogTextBinding
import com.felwal.android.util.string

private const val ARG_TEXT = "text"
private const val ARG_HINT = "hint"

class TextDialog : BaseDialog() {

    private lateinit var listener: DialogListener

    // args
    private lateinit var text: String
    private lateinit var hint: String

    // DialogFragment

    override fun onAttach(c: Context) {
        super.onAttach(c)
        listener = try {
            c as DialogListener
        }
        catch (e: ClassCastException) {
            throw ClassCastException("Activity must implement DialogListener")
        }
    }

    // BaseDialog

    override fun unpackBundle(bundle: Bundle?) {
        bundle?.apply {
            text = getString(ARG_TEXT, "")
            hint = getString(ARG_HINT, "")
        }
    }

    override fun buildDialog(): AlertDialog {
        val binding = DialogTextBinding.inflate(inflater)
        binding.et.inputType = EditorInfo.TYPE_CLASS_TEXT

        binding.et.setText(text)
        binding.et.hint = hint

        return builder.run {
            setView(binding.root)
            setTitle(title)
            if (message != "") setMessage(message)

            setPositiveButton(posBtnTxtRes) { _, _ ->
                val input = binding.et.string.trim { it <= ' ' }
                listener.onTextDialogPositiveClick(input, dialogTag)
            }
            setCancelButton(negBtnTxtRes)

            show()
        }
    }

    //

    interface DialogListener {
        fun onTextDialogPositiveClick(input: String, tag: String)
    }

    //

    companion object {
        @JvmStatic
        fun newInstance(
            title: String,
            message: String = "",
            text: String = "",
            hint: String = "",
            @StringRes posBtnTxtRes: Int = R.string.dialog_btn_ok,
            @StringRes negBtnTxtRes: Int = R.string.dialog_btn_cancel,
            tag: String
        ): TextDialog = TextDialog().apply {
            arguments = putBaseBundle(title, message, posBtnTxtRes, negBtnTxtRes, tag).apply {
                putString(ARG_TEXT, text)
                putString(ARG_HINT, hint)
            }
        }
    }
}

fun textDialog(
    title: String,
    message: String = "",
    text: String = "",
    hint: String = "",
    @StringRes posBtnTxtRes: Int = R.string.dialog_btn_ok,
    @StringRes negBtnTxtRes: Int = R.string.dialog_btn_cancel,
    tag: String
): TextDialog = TextDialog.newInstance(title, message, text, hint, posBtnTxtRes, negBtnTxtRes, tag)